src/TraceEvent/TraceEventSession.cs

```diff
@@ -738,10 +738,30 @@ public unsafe bool EnableKernelProvider(KernelTraceEventParser.Keywords flags, K
                     ETWControl.EnableStackCaching(m_SessionHandle);
                 }
 
+                EnableLastBranchRecordingIfConfigured();
+
                 return ret;
             }
         }
 
+        private unsafe void EnableLastBranchRecordingIfConfigured()
+        {
+            uint[] sources = m_LastBranchRecordingProfileSources;
+            if (sources == null || sources.Length == 0)
+                return;
+
+            uint filters = (uint)m_LastBranchRecordingFilters;
```

Comment:
Can you please add a check here to make sure that this code only runs on supported operating systems?  There is an example of how to do this at [https://github.com/microsoft/perfview/blob/main/src/TraceEvent/TraceEventSession.cs#L1441](https://github.com/microsoft/perfview/blob/main/src/TraceEvent/TraceEventSession.cs#L1441).  In this case, I think throwing an exception that states which operating systems are supported will be a good way to avoid having to investigate issues where LBR support isn't present in the OS.

Reply:
Thanks for adding this @jakobbotsch.  For now, this is fine - if it turns out that we're seeing lots of failures in this area because its being run on old versions, we may need to tighten this up.